dev-java/lz4-java: switch to java-pkg-simple - W I P

vaukai opened the pull request:
- uses java-pkg-simple.eclass instead of java-ant-2.eclass
- Java code generation with mvel translated to ebuild
- compiled java classes are complete, no differences

- headers created from the ebuild using ejavac -h
- shared object file created from the ebuild with gcc https://github.com/lz4/lz4-java/blob/1.8.0/build.xml#L206-L227

- Comparing (abidiff) the shared object file with upstream shows that 100 function symbols are still missing using Makefile from revision -r2 leads to exactly the same result

- Need to find out how to replicate what they do in https://github.com/lz4/lz4-java/blob/1.8.0/build.xml#L222

- Tests to be completed

fordfrog commented:
i looked at loading the native library, i can fix it but so far i didn't find a way to get native library path from java. will try to look at it later, short of time now.

fordfrog commented:
i did some tests and here are my findings:
- the library is found only if it is not installed into a subdirectory but instead directly to the main directory (for example `/usr/lib64`)
- the java code searches for the library in system path so for the tests to work correctly, it should first look for the library in the package work dir to find the newly compiled library, which needs this path to be added for the purpose of the tests

if done this way, the library loads fine and the tests pass. i found this just by linking the library to `/usr/lib64` and emerging the package with `lz4-java` being already installed. i did not tweak the ebuild in any way. so to fix the ebuild, these adjustments need to be done:
- installing the .so file directly in the libdir - this is also needed for runtime, without this change runtime will imo fail too
- passing additional libdir for tests to the newly compiled library - to be done correctly, this should be the first path inspected - this is needed only for the tests to find the correct library

fordfrog commented:
well, there should also be another way. i'm just looking at `tomcat-native` how they load the library and maybe that is a better way to do it.

fordfrog commented:
so i fixed it in 8daa785e9ed2b599d57e8a13d6e064c4fbde5b85

vaukai commented:
> so i fixed it in [8daa785](https://github.com/gentoo/gentoo/commit/8daa785e9ed2b599d57e8a13d6e064c4fbde5b85)

Wow. that's really really cool. 
btw: I've added a chapter [JNI creation only with gcc](https://wiki.gentoo.org/wiki/Java_Developer_Guide/Using_java-pkg-simple.eclass#Examples) to the wiki.

fordfrog commented:
> > so i fixed it in [8daa785](https://github.com/gentoo/gentoo/commit/8daa785e9ed2b599d57e8a13d6e064c4fbde5b85)
> 
> Wow. that's really really cool. btw: I've added a chapter [JNI creation only with gcc](https://wiki.gentoo.org/wiki/Java_Developer_Guide/Using_java-pkg-simple.eclass#Examples) to the wiki.

cool, it's good to have reference for things like these 💪
